oximeter/db/src/client.rs

```diff
@@ -3338,4 +3665,385 @@ mod tests {
             );
         }
     }
+
+    async fn create_test_upgrade_schema_directory(
+        replicated: bool,
+        versions: &[u64],
+    ) -> (TempDir, Vec<PathBuf>) {
+        assert!(!versions.is_empty());
+        let schema_dir = TempDir::new().expect("failed to create tempdir");
+        let mut paths = Vec::with_capacity(versions.len());
+        for version in versions.iter() {
+            let version_dir = schema_dir
+                .path()
+                .join(if replicated { "replicated" } else { "single-node" })
+                .join(version.to_string());
+            fs::create_dir_all(&version_dir)
+                .await
+                .expect("failed to make version directory");
+            paths.push(version_dir);
+        }
+        (schema_dir, paths)
+    }
+
+    #[tokio::test]
```

Comment:
All of these tests currently only run against a single node installation. Now that all tests have been ported over to run against a replicated installation, these should be modified as well

Reply:
Done in e7213592a